textattack/shared/utils/strings.py

```diff
@@ -76,19 +79,43 @@ def words_from_text(s, words_to_ignore=[]):
     # TODO implement w regex
     words = []
     word = ""
-    for c in " ".join(s.split()):
-        if c.isalnum() or c in homos:
-            word += c
-        elif c in "'-_*@" and len(word) > 0:
-            # Allow apostrophes, hyphens, underscores, asterisks and at signs as long as they don't begin the
-            # word.
-            word += c
-        elif word:
-            if word not in words_to_ignore:
-                words.append(word)
-            word = ""
-    if len(word) and (word not in words_to_ignore):
-        words.append(word)
+
+    DetectorFactory.seed = 0
+    try:
+        if s:
+            if detect(s) == "zh-cn" or detect(s) == "ko":
+                seg_list = jieba.cut(s, cut_all=False)
+                s = " ".join(seg_list)
+            else:
+                s = " ".join(s.split())
```

Comment:
Should try/except be made in this session instead to avoid repeating the same code blocks?

Reply:
The reason why it had to be repeated is 

TRY:detect language:

if Chinese/korean, segment using the chinese word segmenter.
         else: use normal segmentation for germanic languages.
EXCEPTION: If language cannot be detected (sometimes language detectors throws an error if the input is empty for example):
          segment the sentence into words.